### ISSUE TITLE
Middleware API lost in npm run build (possible Azure specific)

### ISSUE THREAD

**Issue by gingerhelp** [react-boilerplate/react-boilerplate#2695]
## Description
When adding a middleware api to server/index.js it seems to get lost through the npm run build process.  There is a chance this is an issue specific to Azure.

## Steps to reproduce
Add a custom middleware api to server/index.js.  Mine looks like the following:

```
...
const api = require('./middlewares/api');
...
app.use('/api', api);
...

```

Then my middleware API just has a basic token authentication bones in place:

```
...
router.post('/login', (req, res) => {
  const { username, password } = req.body;
  // For the given username fetch user from DB
  const mockedUsername = 'admin';
  const mockedPassword = 'password';

  if (username && password) {
    if (username === mockedUsername && password === mockedPassword) {
      const token = jwt.sign({ username }, 'worldisfullofdevelopers', {
        expiresIn: '24h', // expires in 24 hours
      });
      // return the JWT token for the future API calls
      res.json({
        success: true,
        message: 'Authentication successful!',
        token,
      });
    } else {
      res.json({
        success: false,
        message: 'Incorrect username or password',
      });
    }
  } else {
    res.json({
      success: false,
      message: 'Authentication failed! Please check the request',
    });
  }
});
...

```

Then upload to an Azure Node.js app service following these steps: https://docs.microsoft.com/en-us/azure/app-service/containers/quickstart-nodejs

**Expected behavior**
- [x] I should be able to use a tool like Postman to interact with this API during development:

```
POST /api/login HTTP/1.1
Host: localhost:3000
Content-Type: application/json
Cache-Control: no-cache
Postman-Token: e56ca744-0af7-f005-3dbd-10a4dec0dbe7

{ 
 "username": "admin",
 "password": "password"
}
```

- [ ] I should be able to use Postman to perform this same API call to the deployed build code:

```
POST /api/login HTTP/1.1
Host: MYSERVERNAME.azurewebsites.net
Content-Type: application/json
Cache-Control: no-cache
Postman-Token: bd42e331-5c47-dc15-b8a0-a59f14f64295

{ 
 "username": "admin",
 "password": "password"
}
```

Error: The resource you are looking for has been removed, had its name changed, or is temporarily unavailable.

## Versions

- React-Boilerplate: 4.0.0
- Node/NPM: 10.13.0 locally
- Browser: N/A

**Comment by julienben** [comment]
How is Azure serving your code? Do you deploy to a node server and then run `npm run start:production`? If you don't, then Azure is serving your code by itself without using your express server at all. (That's the most common case for deploying an SPA IME.)

**Comment by gingerhelp** [comment]
I believe it serves the code up by itself until you put in a web.config that tells it to use iisnode instead.  In that link I had with the steps for publish a Node app to Azure they reference a GitHub project at https://github.com/Azure-Samples/nodejs-docs-hello-world and it has a web.config within per the following:

```
<?xml version="1.0" encoding="utf-8"?>
<!--
     This configuration file is required if iisnode is used to run node processes behind
     IIS or IIS Express.  For more information, visit:

     https://github.com/tjanczuk/iisnode/blob/master/src/samples/configuration/web.config
-->

<configuration>
  <system.webServer>
    <!-- Visit http://blogs.msdn.com/b/windowsazure/archive/2013/11/14/introduction-to-websockets-on-windows-azure-web-sites.aspx for more information on WebSocket support -->
    <webSocket enabled="false" />
    <handlers>
      <!-- Indicates that the server.js file is a node.js site to be handled by the iisnode module -->
      <add name="iisnode" path="index.js" verb="*" modules="iisnode"/>
    </handlers>
    <rewrite>
      <rules>
        <!-- Do not interfere with requests for node-inspector debugging -->
        <rule name="NodeInspector" patternSyntax="ECMAScript" stopProcessing="true">
          <match url="^index.js\/debug[\/]?" />
        </rule>

        <!-- First we consider whether the incoming URL matches a physical file in the /public folder -->
        <rule name="StaticContent">
          <action type="Rewrite" url="public{REQUEST_URI}"/>
        </rule>

        <!-- All other URLs are mapped to the node.js site entry point -->
        <rule name="DynamicContent">
          <conditions>
            <add input="{REQUEST_FILENAME}" matchType="IsFile" negate="True"/>
          </conditions>
          <action type="Rewrite" url="index.js"/>
        </rule>
      </rules>
    </rewrite>
    
    <!-- 'bin' directory has no special meaning in node.js and apps can be placed in it -->
    <security>
      <requestFiltering>
        <hiddenSegments>
          <remove segment="bin"/>
        </hiddenSegments>
      </requestFiltering>
    </security>

    <!-- Make sure error responses are left untouched -->
    <httpErrors existingResponse="PassThrough" />

    <!--
      You can control how Node is hosted within IIS using the following options:
        * watchedFiles: semi-colon separated list of files that will be watched for changes to restart the server
        * node_env: will be propagated to node as NODE_ENV environment variable
        * debuggingEnabled - controls whether the built-in debugger is enabled

      See https://github.com/tjanczuk/iisnode/blob/master/src/samples/configuration/web.config for a full list of options
    -->
    <!--<iisnode watchedFiles="web.config;*.js"/>-->
  </system.webServer>
</configuration>
```

**Comment by gingerhelp** [comment]
Some additional insight - I just signed up for a Heroku account and everything works perfectly there so this is definitely an Azure specific issue.  Perhaps there are feature limitations to iisnode coming into play here as well.  I am going to try to create the resource out there a bit differently and see if I can get a pure node environment as opposed to iisnode - will report back.

**Comment by gingerhelp** [comment]
Ok, confirming it here - the issue is resolved if the correct type of resource is created on Azure.  So for any future folks searching for this, DO NOT follow the instructions posted at https://docs.microsoft.com/en-us/azure/app-service/containers/quickstart-nodejs if you are looking to host your react-boilerplate project.  Instead, manually create a resource from their marketplace listed as 'Web App' and select one of the node versions for the runtime stack.  Then I deployed using local Git by selecting that from the Deployment Center, set up my user name and password, and then ran the following commands from the root of the source folder (not the build subfolder):

```
git remote add azure https://MYSITE.scm.azurewebsites.net:443/MYSITE.git
git push azure master
```

I saw all of the same scripts running (including the npm run build) as on Heroku using this methodology.  I will do a pull request to update the documentation on deployment to add an entry for Azure.
